src/main/java/com/brewmes/demo/model/Report.java

```diff
@@ -0,0 +1,312 @@
+		if (currentBatch.getTemperature().keySet().stream().findFirst().isPresent()) {
+			LocalDateTime startTime;
+			startTime = currentBatch.getTemperature().keySet().stream().findFirst().get();
+
+			for (LocalDateTime time : currentBatch.getTemperature().keySet()) {
+				dataset.addValue(currentBatch.getTemperature().get(time), "Temperature",
+						Math.abs(startTime.toEpochSecond(ZoneOffset.MAX) - time.toEpochSecond(ZoneOffset.MAX)) + "");
+			}
+		}
+        JFreeChart lineChart = ChartFactory.createLineChart("Temperature", "Time", "Temperature"
+                , dataset, PlotOrientation.VERTICAL, true, true, false);
+        CategoryPlot plot = lineChart.getCategoryPlot();
+        CategoryAxis xAis = lineChart.getCategoryPlot().getDomainAxis();
+        xAis.setCategoryLabelPositions(CategoryLabelPositions.UP_90);
+
+        PdfContentByte contentByte = pdfWriter.getDirectContent();
+        PdfTemplate template = contentByte.createTemplate(width, height);
+        Graphics2D graphics2d = template.createGraphics(width, height, new DefaultFontMapper());
+        Rectangle2D rectangle2d = new Rectangle2D.Double(0, 0, width, height);
+
+        lineChart.draw(graphics2d, rectangle2d);
+        graphics2d.dispose();
+        Image chartImage = Image.getInstance(template);
+        temperature.add(chartImage);
+
+        // Create the table for temperature
+        PdfPTable table = new PdfPTable(3);
+        table.addCell("Minimum");
+        table.addCell("Maximum");
+        table.addCell("Average");
+        table.addCell("?");
+        table.addCell("?");
+        table.addCell("?");
+        // Add table to document underneath the graph
+        addEmptyLine(temperature, 1);
+        temperature.add(table);
+
+        document.add(temperature);
+    }
+
+    public static void addTimeSection(Document document) throws DocumentException {
+        Paragraph timeState = new Paragraph();
+        addEmptyLine(timeState, 1);
+
+        int width = 500;
+        int height = 400;
+
+        // Create the bar chart for time in states
+        DefaultCategoryDataset dataset = new DefaultCategoryDataset();
+        dataset.setValue(2, "state", "Deactivated");
+        dataset.setValue(1, "state", "Clearing");
+        dataset.setValue(1, "state", "Stopped");
+        dataset.setValue(4, "state", "Starting");
+        dataset.setValue(0, "state", "Idle");
+        dataset.setValue(0, "state", "Suspended");
+        dataset.setValue(7, "state", "Execute");
+        dataset.setValue(10, "state", "Stopping");
+        dataset.setValue(2, "state", "Aborting");
+        dataset.setValue(2, "state", "Aborted");
+        dataset.setValue(2, "state", "Holding");
+        dataset.setValue(5, "state", "Resetting");
+        dataset.setValue(2, "state", "Completing");
+        dataset.setValue(2, "state", "Complete");
+        dataset.setValue(2, "state", "Deactivating");
+        dataset.setValue(8, "state", "Activating");
```

Comment:
Consider making "state" a variable, as to not write the same thing 17 times.

Reply:
getting fixed when rebased :)